File: app/src/main/java/com/example/a59682/myapplicationtest2018421/UploadActivity.java
package com.example.a59682.myapplicationtest2018421;

import android.content.ContentResolver;
import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.net.Uri;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

import java.io.FileNotFoundException;

/**
 * @author 59682
 * @name UploadActivity
 * @description 图片上传
 */
public class UploadActivity extends AppCompatActivity {

    private Button Browse;
    private ImageView Images;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_upload);

        /* 设置toolbar */
        Toolbar toolbar = findViewById(R.id.upload_toolbar);
        setSupportActionBar(toolbar);

        /* 设置bar左侧按钮 */
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
            //设置导航按钮
            //actionBar.setHomeAsUpIndicator(R.mipmap.navigation);
        }

        init();

        /* 点击打开UpLoadImage，使用Album进行上传 */
        Browse.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent();
                intent.setClass(UploadActivity.this, UploadImage.class);
                startActivity(intent);
            }
        });
    }

    /* 初始化 */
    private void init() {
        Browse = (Button) findViewById(R.id.browse);
        Images = (ImageView) findViewById(R.id.camera_image);
    }

    /* 选取并展示 ？ */
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        if (requestCode == RESULT_OK) {
            Uri uri = data.getData();
            ContentResolver cr = this.getContentResolver();
            try {//压缩图片
                Bitmap mBitmap = BitmapFactory.decodeStream(cr.openInputStream(uri));
            } catch (FileNotFoundException e) {
                e.printStackTrace();
            }
        }
        super.onActivityResult(requestCode, resultCode, data);
    }

    /**
     * @description 设置menu点击事件
     */
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            /* 返回键 */
            case R.id.backup:
                Intent intent_backup = new Intent();
                intent_backup.setClass(UploadActivity.this, LoginActivity.class);
                startActivity(intent_backup);
                /* web键 */
            case R.id.web:
                Intent intent_web = new Intent();
                intent_web.setClass(UploadActivity.this, WebActivity.class);
                startActivity(intent_web);
                /* home键 */
            case R.id.home:
                Intent intent_home = new Intent();
                intent_home.setClass(UploadActivity.this, BackgroundActivity.class);
                startActivity(intent_home);
        }
        return true;
    }
}

File: app/src/main/java/com/ten/lifecat/phone/view/activity/UploadActivity.kt
package com.ten.lifecat.phone.view.activity

import android.app.Activity
import android.content.Intent
import android.graphics.BitmapFactory
import android.os.Bundle
import android.view.MenuItem
import android.view.View
import android.widget.Button
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.ten.lifecat.phone.R
import java.io.FileNotFoundException

/**
 * 图片上传界面
 */
class UploadActivity : AppCompatActivity() {

    private var Browse: Button? = null
    private var Images: ImageView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_upload)

        /* 设置toolbar */
        val toolbar = findViewById<Toolbar>(R.id.upload_toolbar)
        setSupportActionBar(toolbar)

        /* 设置bar左侧按钮 */
        val actionBar = supportActionBar
        actionBar?.setDisplayHomeAsUpEnabled(true)

        init()

        /* 点击打开UpLoadImage，使用Album进行上传 */
        Browse!!.setOnClickListener {
            startActivity(Intent(this, UploadImage::class.java))
        }
    }

    /* 初始化 */
    private fun init() {
        Browse = findViewById(R.id.browse)
        Images = findViewById<View>(R.id.camera_image) as ImageView
    }

    /* 选取并展示 ？ */
    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        if (requestCode == Activity.RESULT_OK) {
            val uri = data!!.data
            val cr = this.contentResolver
            try {//压缩图片
                val mBitmap = BitmapFactory.decodeStream(cr.openInputStream(uri!!))
            } catch (e: FileNotFoundException) {
                e.printStackTrace()
            }

        }
        super.onActivityResult(requestCode, resultCode, data)
    }

    /**
     * 设置menu点击事件
     */
    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            /* 返回键 */
            R.id.backup -> {
                startActivity(Intent(this, LoginActivity::class.java))
            }
            /* web键 */
            R.id.web -> {
                startActivity(Intent(this, WebActivity::class.java))
            }
            /* home键 */
            R.id.home -> {
                startActivity(Intent(this, BackgroundActivity::class.java))
            }
            else -> {
            }
        }
        return true
    }
}
